### PR TITLE
Add Scatter to Topi/Relay/ONNX via hybrid script

### PR DESCRIPTION
**Pull request by mbrookhart**
cc: @jwfromm @tqchen @masahi 

The actual hybrid script implementation is kind of gross, I can't seem to figure out a way to clean it up without causing very opaque errors in the hybrid script module. If anyone has suggestions on cleaning that up, I'd love to hear them.

### CODE REVIEW

**Comment by mbrookhart** [comment]
@jwfromm @masahi @Laurawly @Huyuwei I finally found time to get back to this and fix the issues with GPU/i386. Could you take a look? Thanks!

File: src/relay/op/tensor/transform.cc [hunk 1]
```diff
@@ -781,6 +781,53 @@ non-zero)doc" TVM_ADD_FILELINE)
...
+RELAY_REGISTER_OP("scatter")
+    .describe(
+        R"doc(Update data at positions defined by indices with values in updates)doc" TVM_ADD_FILELINE)
+    .set_num_inputs(3)
+    .add_argument("data", "Tensor", "The input data tensor.")
+    .add_argument("indicies", "Tensor", "The indicies location tensor.")
+    .add_argument("updates", "Tensor", "The values to update the input with.")
+    .add_type_rel("Scatter", ScatterRel)
+    .set_attr<TOpIsStateful>("TOpIsStateful", false)
+    .set_attr<TOpPattern>("TOpPattern", kOpaque)
```
Comment (masahi) [inline src/relay/op/tensor/transform.cc:828]: Why opaque?

**Comment by notoraptor** [comment]
Hi @mbrookhart ! Is this implementation of `scatter` the same as the one in pytorch here ? https://pytorch.org/docs/stable/tensors.html#torch.Tensor.scatter_

As I implemented pytorch's `gather` in #5716 , I planned to implement reverse operation `scatter` too.

File: src/relay/op/tensor/transform.cc (hunk 1 above)
Comment (mbrookhart) [reply src/relay/op/tensor/transform.cc:828]: Because I washing sure and I wanted to be conservative :) It might be Injective.

**Comment by mbrookhart** [comment]
@notoraptor I based it off ONNX, which seems to be doing the same thing as the Pytorch op, if I'm reading the docs correctly. So, I think so, yes.

File: src/relay/op/tensor/transform.cc (hunk 1 above)
Comment (masahi) [reply src/relay/op/tensor/transform.cc:827]: ok for now

**Comment by masahi** [comment]
@mbrookhart There is some issue with WASM build, can you try again?

cc @tqchen https://ci.tvm.ai/blue/organizations/jenkins/tvm/detail/PR-5619/9/pipeline

**Comment by masahi** [comment]
Thanks @mbrookhart

**Pull request merged**